docs/user_manual/components.md

```diff
@@ -695,15 +695,116 @@ A `fault` has no steady state output.
+| `u_set`                    | `double`                                                                                                                                                                                                                                                                                                           | volt (V) | the voltage setpoint (at the center of the band)                                                        | &&#10024; only for power flow with automatic tap changing | &#10004; |      `>= 0`       |
+| `u_band`                   | `double`                                                                                                                                                                                                                                                                                                           | volt (V) | the width of the voltage band ($=2*\left(\Delta U\right)_{\text{acceptable}}$)                          | &&#10024; only for power flow with automatic tap changing | &#10004; | `> 0` (see below) |
+| `line_drop_compensation_r` | `double`                                                                                                                                                                                                                                                                                                           | ohm (Ω)  | compensation for voltage drop resistance during transport (see [below](#line-drop-compensation))        |                  &#10060; default `0.0`                   | &#10004; |      `>= 0`       |
+| `line_drop_compensation_x` | `double`                                                                                                                                                                                                                                                                                                           | ohm (Ω)  | compensation for voltage drop due to  reactance during transport (see [below](#line-drop-compensation)) |                  &#10060; default `0.0`                   | &#10004; |      `>= 0`       |
+
+The following additional requirements exist on the input parameters.
+
+- The `control_side` supports {py:class}`BranchSide <power_grid_model.enum.BranchSide>` if it regulates a {hoverxreftooltip}`user_manual/components:transformer` and {py:class}`Branch3Side <power_grid_model.enum.Branch3Side>` if it regulates a {hoverxreftooltip}`user_manual/components:Three-Winding Transformer`.
+- The rated voltage at the `tap_side` must be greater than or equal to the rated voltage at the `control_side`.
```

Comment:
```suggestion
- The automatic tap changer algorithm currently only supports tap changers connected at HV side of the transformer. Hence, the rated voltage of node at the `tap_side` must be greater than or equal to the rated voltage of node at the other side of transformer.
```

Control side can be both HV or LV. But only tap_side can be HV.

Reply:
i resolved this, but semi-related: what's the difference between `u1`/`u2` of the transformers and the `u_rated` of the nodes they are connected to?

Reply:
All these 4 can be different values. `u1`, `u2` are about the actual transformer parameters. They would usually determine the transformation ratio of voltage.

`u_rated` usually is almost equal to `u1` or `u2`. But it does not have to be equal. All the associated `%` or p.u. input grid parameters get affected by the `u_rated` voltages. You can attach any `u1`/`u2` transformer to any `u_rated_1` / `u_rated_2`

While manufacturers could indeed combine all these and consider only 1 value or 2  values, they remain separate as they are all linked to other parameters individually within & outside of power flow calculations.

Also, we could have done ranking against `u1` or `u2`, but that can potentially give more groups of ranks. Different brands of transformer might have different ratings: like maybe some have `10000` or some have `10050` (There might be better examples for this)

**TLDR**: And in relevance of this comment, the check we do is against u_rated of both nodes.

Reply:
great writeup. thanks!